restapi/src/test/java/io/stargate/web/docsapi/service/QueryExecutorTest.java

```diff
@@ -414,4 +422,160 @@ void testSubDocumentsPaged(int pageSize) {
       assertThat(docs).isEmpty();
     }
   }
+
+  @ParameterizedTest
+  @CsvSource({"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "100"})
+  void testMergeByDocKey(int pageSize) {
```

Comment:
can we have the same extended depth please?

Reply:
will add

Reply:
tests added in the new set of commits